python/ray/tune/search/optuna/optuna_search.py

```diff
@@ -602,34 +602,23 @@ def add_evaluated_point(
         self._ot_study.add_trial(trial)
 
     def save(self, checkpoint_path: str):
-        save_object = (
-            self._sampler,
-            self._ot_trials,
-            self._ot_study,
-            self._points_to_evaluate,
-            self._evaluated_rewards,
-        )
+        save_object = self.__dict__.copy()
         with open(checkpoint_path, "wb") as outputFile:
             pickle.dump(save_object, outputFile)
 
     def restore(self, checkpoint_path: str):
         with open(checkpoint_path, "rb") as inputFile:
             save_object = pickle.load(inputFile)
-        if len(save_object) == 5:
-            (
-                self._sampler,
-                self._ot_trials,
-                self._ot_study,
-                self._points_to_evaluate,
-                self._evaluated_rewards,
-            ) = save_object
+        if isinstance(save_object, dict):
```

Comment:
I've kept one level of backwards compatibility. For example, optuna searcher load was checking for length 5 state tuple vs. length 4. We don't really expect users to restore searchers across versions - would it be fine to remove these backwards compatibility cases fully?

Reply:
Let's mark this for removal (either as a comment or in a tracker sheet). I don't think it's a hard requirement but let's keep it for now!